app/models/app_filter.rb

```diff
@@ -1,7 +1,6 @@
 # frozen_string_literal: true
 class AppFilter < ApplicationRecord
-  belongs_to :app
-  validates_associated :app
+  belongs_to :app, :optional => true
```

Comment:
When we want a filter on all apps, we won't create any `AppFilter` record. The same goes for any other types of connection tables.
When we will match the filter, we will use left outer join with these tables. It will produce either fields with values, or empty fields, in case there was no single record present. We can check then for either desired value, or a `null` value.

Reply:
I don't see a reason for 3 - just disable the rule. I think we can drop it, and make the code simpler.
In the simple case it should be even possible to express the whole query using rails methods, without resolving to writing SQL in code.